pkg/autoscaler/stats_scraper.go

```diff
@@ -45,9 +45,18 @@ const (
 	// to retry if a Scrape returns an error or if the Scrape goes to a pod we already
 	// scraped.
 	scraperMaxRetries = 10
+
+	// If the sample size is smaller than this, then we don't consider age.
+	minSampleSizeToConsiderAge = 4
 )
 
 var (
+	// Pod older than 1 minute is considered old enough for its
+	// stat to be included in the sample immediately.
+	// This number is arbitrary decided to be equal to the default
+	// stable window.
+	youngPodCutOffSecs = time.Minute.Seconds()
```

Comment:
This feels like it should be relative to the stable window size 🤔

Reply:
I can do this in addition. It's not a hard change, but requires lots of interface manipulations. Can send it as a follow up :)